featuretools/tests/utils_tests/test_entity_utils.py

```diff
@@ -0,0 +1,101 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import, division, print_function
+
+import pytest
+
+import featuretools as ft
+from featuretools import variable_types as vtypes
+from featuretools.utils.entity_utils import (
+    convert_all_variable_data,
+    convert_variable_data,
+    get_linked_vars,
+    infer_variable_types
+)
+
+
+@pytest.fixture(scope='module')
+def es():
+    es = ft.demo.load_mock_customer(return_entityset=True)
+    return es
+
+
+def test_infer_variable_types(es):
+    # Customers Entity
+    total_variables = es['customers'].df.columns
+    variable_types = ['customer_id']
+
+    entity = es['customers']
+    link_vars = get_linked_vars(entity)
+    inferred_variable_types = infer_variable_types(entity.df,
+                                                   link_vars,
+                                                   variable_types,
+                                                   'join_date',
+                                                   {})
+
+    # Check columns' number
+    assert len(variable_types) + len(inferred_variable_types) == len(total_variables)
+
+    # Check columns' types
+    assert isinstance(entity['join_date'], inferred_variable_types['join_date'])
+    assert inferred_variable_types['join_date'] == vtypes.Datetime
+
+    assert isinstance(entity['date_of_birth'], inferred_variable_types['date_of_birth'])
+    assert inferred_variable_types['date_of_birth'] == vtypes.Datetime
+
+    assert isinstance(entity['zip_code'], inferred_variable_types['zip_code'])
+    assert inferred_variable_types['zip_code'] == vtypes.Categorical
+
+    # Sessions Entity
+    entity = es['sessions']
+    link_vars = get_linked_vars(entity)
+    total_variables = entity.df.columns
+    variable_types = ['session_id']
+    inferred_variable_types = infer_variable_types(entity.df,
+                                                   link_vars,
+                                                   variable_types,
+                                                   'session_start',
+                                                   {})
+
+    # Check columns' number
+    assert len(variable_types) + len(inferred_variable_types) == len(total_variables)
+
+    # Check columns' types
+    assert inferred_variable_types['customer_id'] == vtypes.Ordinal
+
+    assert isinstance(entity['device'], inferred_variable_types['device'])
+    assert inferred_variable_types['device'] == vtypes.Categorical
+
+    assert isinstance(entity['session_start'], inferred_variable_types['session_start'])
+    assert inferred_variable_types['session_start'] == vtypes.Datetime
+
+
+def test_convert_all_variable_data(es):
```

Comment:
this test doesn't clearly test that the variable data was actually converted. 

when you rewrite this test there is no need to test the variable types multiple times (e.g Numeric is tested 4 times), but each variable type should be tested once